test/000_fullchain.test.ts

```diff
@@ -198,6 +206,97 @@ describe('Integration tests', function () {
+                const schedulerStakePerTask = schedulerStakePerDeal / volume;
+                const workerRewardPerTask = await iexecWrapper.computeWorkerRewardPerTask(
+                    dealId,
+                    PocoMode.CLASSIC,
+                );
+                const schedulerRewardPerTask = workerpoolPrice - workerRewardPerTask;
+                // Check initial balances.
+                let accountsInitBalances = [
+                    {
+                        address: proxyAddress,
+                        balance: (await iexecPoco.balanceOf(proxyAddress)).toNumber(),
+                        frozen: (await iexecPoco.frozenOf(proxyAddress)).toNumber(),
+                    },
+                ];
+                for (const account of acounts) {
+                    let address = account.address;
+                    let balance = (await iexecPoco.balanceOf(account.address)).toNumber();
+                    let frozen = (await iexecPoco.frozenOf(account.address)).toNumber();
+                    accountsInitBalances.push({ address, balance, frozen });
+                }
+                for (let i = 0; i < workerNumber; i++) {
+                    expect(await iexecPoco.viewScore(workers[i].address)).to.be.equal(0);
+                }
+                const taskId = await iexecWrapper.initializeTask(dealId, 0);
+                // Finalize each task and check balance changes.
+                const workerStake = await iexecPoco
+                    .viewDeal(dealId)
+                    .then((deal) => deal.workerStake.toNumber());
+
+                for (let i = 0; i < workerNumber; i++) {
+                    await iexecWrapper.contributeToTask(dealId, 0, resultDigest, workers[i]);
+                }
+                for (let i = 0; i < workerNumber; i++) {
+                    await iexecPoco
+                        .connect(workers[i])
+                        .reveal(taskId, resultDigest)
+                        .then((tx) => tx.wait());
+                }
+                await iexecPoco
+                    .connect(scheduler)
+                    .finalize(taskId, results, '0x')
+                    .then((tx) => tx.wait());
+                expect((await iexecPoco.viewTask(taskId)).status).to.equal(
+                    TaskStatusEnum.COMPLETED,
+                );
+                const expectedBalanceChanges = [
+                    -(dealPrice + schedulerStakePerDeal),
+                    0,
+                    schedulerStakePerTask + schedulerRewardPerTask,
+                    appPrice,
+                    datasetPrice,
+                ];
+                const expectedFrozenChanges = [0, -taskPrice, -schedulerStakePerTask, 0, 0];
+                for (let i = 0; i < workerNumber; i++) {
+                    expectedBalanceChanges.push(workerStake + workerRewardPerTask / workerNumber);
+                    expectedFrozenChanges.push(0);
+                }
+                await changesInBalancesAndFrozens({
```

Comment:
Why not using
```
expect(tx).to.changeTokenBalances(iexecPoco, [iexecPoco, requester, scheduler, appProvider, datasetProvider, [
      -(dealPrice + schedulerStakePerDeal),
      0,
      schedulerStakePerTask + schedulerRewardPerTask,
      appPrice,
      datasetPrice,
  ])
 for (const worker of workers){
    expect(tx).to.changeTokenBalance(iexecPoco, worker, workerStake + workerRewardPerTask / workerNumber)
}
```
then your custom method
`await expectToChangeFrozen(..)`
?

Reply:
I can if you prefer yes